pkg/manager/member/scaler.go

```diff
@@ -179,12 +181,32 @@ func ordinalPodName(memberType v1alpha1.MemberType, tcName string, ordinal int32
 // - ordinal: pod ordinal to create or delete
 // - replicas/deleteSlots: desired replicas and deleteSlots by allowing only one pod to be deleted or created
 func scaleOne(actual *apps.StatefulSet, desired *apps.StatefulSet) (scaling int, ordinal int32, replicas int32, deleteSlots sets.Int32) {
+	var ordinals []int32
+	scaling, ordinals, replicas, deleteSlots = scaleMulti(actual, desired, 1)
+	if len(ordinals) == 0 {
+		ordinal = -1
+	} else {
+		ordinal = ordinals[0]
+	}
+	return
+}
+
+// scaleMulti calculates desired replicas and delete slots from actual/desired
+// StatefulSets by allowing multiple pods to be deleted or created
+// it returns following values:
+// - scaling:
+//   - 0: no scaling required
+//   - 1: scaling out
+//   - -1: scaling in
+// - ordinals: pod ordinals to create or delete
+// - replicas/deleteSlots: desired replicas and deleteSlots by allowing no more than maxCount pods to be deleted or created
+func scaleMulti(actual *apps.StatefulSet, desired *apps.StatefulSet, maxCount int) (scaling int, ordinals []int32, replicas int32, deleteSlots sets.Int32) {
```

Comment:
Define a struct as return result maybe clearer

Reply:
Just keep the same with origin `scaleOne` here. 

Do we have to do this refactor? (both `scaleOne` and `scaleMulti` should be considered together I think ?

Reply:
> Just keep the same with origin `scaleOne` here.
> 
> Do we have to do this refactor? (both `scaleOne` and `scaleMulti` should be considered together I think ?

@DanielZhangQD

Reply:
@lizhemingi @liubog2008 Let's refactor this later.

Reply:
lgtm